### DISCUSSION TITLE
ef.stereo_width as expander

### DISCUSSION THREAD

**Discussion by jpcima** [grame-cncm/faustlibraries#96] [Ideas]
The width effect at `ef.stereo_width` accepts w=1 for no change, and w<1 for width reduction.
It appears that this effect may work well for widening (w>1) as well.

This, however, boosts the volume greatly. May there be a version of `stereo_width` which supports widening, and have a compensating attenuation at the output?

**Comment by sletz** [comment]
Any PR to suggest?

**Reply by jpcima** [reply]
No, in truth I am bugged by another issue that comes up.
When trying to harmonize the volume across the range, I'm noticing the range 0-1 as already problematic;
there is a decrease of volume towards the value 1.
This perception of volume variation up->down->up is a little upsetting.

**Comment by josmithiii** [comment]
For "widening", you want "stereo cross-cancellation", which is a different effect.  I believe Manfred Schroeder published the first paper I've seen on this.

Your volume change is caused by amplitude cancellation when you sum left and right to get mono (w=0).
You might try negating one of the channels to see if that helps.
A more sophisticated method (used for seamless looping in samplers, e.g.) is to align phase at all frequencies so that no cancellation occurs.  When this is done, the amplitude will stay precisely the same as a function of w (0 to 1) for any listener at the alignment "sweet spot".
